### DISCUSSION TITLE
Add ResultCallback in types.Options and move package runner outside package internal

### DISCUSSION THREAD

**Discussion by 0xdu** [projectdiscovery/nuclei#2472] [Q&A] [answered]
### Describe the use case of this feature:
This is very useful when integrating nuclei with other tools to get results

For example:

```go
type Options struct {
	//other options
    .........
    ResultCallback func(result *output.ResultEvent)
}
```

When a vulnerability is discovered, this option will be called like:
```go
if (runner.Options.ResultCallback != nil) {
       runner.Options.ResultCallback(issue)
}
```

**Comment by Ice3man543** [comment, marked as answer]
@lucy9x this is already possible using a custom output writer that satisfies output.Writer interface. 

```go
outputWriter := testutils.NewMockOutputWriter()
outputWriter.WriteCallback = func(event *output.ResultEvent) {
	fmt.Printf("Got Result: %v\n", event)
}
```

Regarding making runner public, that's not being done due to obvious reasons since we don't want to make the interface static and part of nuclei public API

**Reply by 0xdu** [reply]
Thanks for the information. Also I have a question, Do you guys plan to upgrade the version of goflag from 0.0.8 to the latest in this project? I see a big difference between version 0.08 and 0.09 of the goflag package

**Reply by forgedhallpass** [reply]
@lucy9x [goflags](https://github.com/projectdiscovery/goflags) is also our project. I wouldn't say that there are that many [changes](https://github.com/projectdiscovery/goflags/releases/tag/v0.0.9). In any case, we are trying to keep all of our dependencies up-to-date, but in some cases it might take a short while.
